server/storage/sqldb.go

```diff
@@ -68,19 +99,25 @@ func (db *SQLStorage) UpdateCurrent(gun data.GUN, update MetaUpdate) error {
 
 	if err := func() error {
 		// write new TUFFile entry
-		if err = translateOldVersionError(tx.Create(&TUFFile{
+		tufFile := TUFFile{
 			Gun:     gun.String(),
 			Role:    update.Role.String(),
 			Version: update.Version,
 			SHA256:  hexChecksum,
 			Data:    update.Data,
-		}).Error); err != nil {
+		}
+		if err = translateOldVersionError(tx.Create(&tufFile).Error); err != nil {
+			return err
+		}
+
+		// The Channel is added separately from the TUFFile record to prevent gorm from issuing an Update
+		if err = tx.Model(&tufFile).Association("Channels").Append(update.Channels).Error; err != nil {
```

Comment:
Good point, I will add a test for it.

Getting gorm to generate the right SQL for this pr took me the better part of a day, I was almost ready to rip it out and replace it...but of course that would've made this much harder to review.